### ISSUE TITLE
Style refactor all .cpp filenames + adjust license formatting for clang-format

### ISSUE THREAD

**Issue by anonimal** [monero-project/kovri#280]
**By submitting this issue, I confirm the following:**

- I have read and understood the [contributor guide](https://github.com/monero-project/kovri/blob/master/doc/CONTRIBUTING.md).
- I have checked that the issue I am reporting can be replicated or that the feature I am suggesting is not present.
- I have checked opened or recently closed [pull requests](https://github.com/monero-project/kovri/pulls) for existing solutions/implementations to my issue/suggestion.

*Place an X inside the bracket to confirm*
- [X] I confirm.

---

1. To keep in order with our upstream guide, we should rename all files from ```.cpp``` to ```.cc``` and change from ```MixedCase.cpp``` to ```single_case_with_underscore.cc```. Once that's done, remove the line about filename style in our style guide.

2. Because our column limit for clang-format is 80, clang-format will squish the license in formatted files. To avoid this, we need to amend a ```//``` at the end of each line like so:
 
```cpp
/**                                                                                       //
 * Copyright (c) 2015-2016, The Kovri I2P Router Project                                  //
 *                                                                                        //
 * All rights reserved.                                                                   //
 *                                                                                        //
 * Redistribution and use in source and binary forms, with or without modification, are   //
 * permitted provided that the following conditions are met:                              //
 *                                                                                        //
 * 1. Redistributions of source code must retain the above copyright notice, this list of //
 *    conditions and the following disclaimer.                                            //
 ```

or, amend ```clang-format on``` ```clang-format off``` before and after the license which I think is uglier and intrusive.  I propose we stick with the less-intrusive version of ```//```. If there is a better way to accomplish 2., I'm all ears.

Referencing #279 and #238.

**Comment by anonimal** [comment]
Unassigned myself. [guzzi jones](https://github.com/guzzijones) said he'd take the issue.

**Comment by EinMByte** [comment]
Let's stick with appending `//`.
Use of `.cc` is somewhat less common, but I'm fine with it. Lowercase/underscore has the advantage of avoiding compilation errors (as we've had) on case insensitive file systems, but is a bit less intuitive IMHO. That said, I don't really mind the change (these are not important matters anyway).

**Comment by anonimal** [comment]
Well, it all adds up (many smaller 'not important' matters turn into large important matter) so I'm glad this is being taken care of. :+1: on the case insensitivity solution; we can resolve that NetDb filename issue once this is resolved.

**Comment by anonimal** [comment]
Resolved with #299 and #302.

Thanks @guzzijones.

**Comment by anonimal** [comment]
@EinMByte JFTR: Boost (tested on 1.60) is still problematic with ```netdb.h``` and will fail to compile on Arch Linux x86_64 - even with include guard in place (see also ```/usr/include/netdb.h```).

For now, we'll keep ```net_db.h``` to complement ```net_db_requests.h``` but I would like to see how this conflict can be resolved (or if it has been resolved).
